packages/in_app_purchase/in_app_purchase_storekit/example/shared/RunnerTests/InAppPurchasePluginTests.m

```diff
@@ -332,69 +354,69 @@ - (void)testAddPaymentSuccessWithPaymentDiscount {
-  OCMStub([mockHandler addPayment:[OCMArg any]]).andReturn(YES);
-  self.plugin.paymentQueueHandler = mockHandler;
+  TestPaymentQueueHandler *handler = [[TestPaymentQueueHandler alloc] init];
+  self.plugin.paymentQueueHandler = handler;
+
+  __block NSInteger addPaymentTimes = 0;
+  handler.addPaymentStub = ^(SKPayment *payment) {
+    if (@available(iOS 12.2, *)) {
+      SKPaymentDiscount *discount = payment.paymentDiscount;
+      XCTAssertEqual(discount.identifier, @"test_identifier");
+      XCTAssertEqual(discount.keyIdentifier, @"test_key_identifier");
+      XCTAssertEqualObjects(
+          discount.nonce,
+          [[NSUUID alloc] initWithUUIDString:@"4a11a9cc-3bc3-11ec-8d3d-0242ac130003"]);
+      XCTAssertEqual(discount.signature, @"test_signature");
+      addPaymentTimes++;
+      return YES;
+    }
+    addPaymentTimes++;
+    return YES;
+  };
 
   FlutterError *error;
 
   [self.plugin addPaymentPaymentMap:argument error:&error];
+  XCTAssertEqual(addPaymentTimes, 1);
   XCTAssertNil(error);
-  OCMVerify(
-      times(1),
-      [mockHandler
-          addPayment:[OCMArg checkWithBlock:^BOOL(id obj) {
-            SKPayment *payment = obj;
-            if (@available(iOS 12.2, *)) {
-              SKPaymentDiscount *discount = payment.paymentDiscount;
-
-              return [discount.identifier isEqual:@"test_identifier"] &&
-                     [discount.keyIdentifier isEqual:@"test_key_identifier"] &&
-                     [discount.nonce
-                         isEqual:[[NSUUID alloc]
-                                     initWithUUIDString:@"4a11a9cc-3bc3-11ec-8d3d-0242ac130003"]] &&
-                     [discount.signature isEqual:@"test_signature"] &&
-                     [discount.timestamp isEqual:@(1635847102)];
-            }
-
-            return YES;
-          }]]);
 }
 
 - (void)testAddPaymentFailureWithInvalidPaymentDiscount {
   // Support for payment discount is only available on iOS 12.2 and higher.
   if (@available(iOS 12.2, *)) {
-    NSDictionary *argument = @{
+    NSDictionary *invalidDiscount = @{
       @"productIdentifier" : @"123",
       @"quantity" : @(1),
       @"simulatesAskToBuyInSandbox" : @YES,
       @"paymentDiscount" : @{
+        /// This payment discount is missing the field `identifier`, and is thus malformed
         @"keyIdentifier" : @"test_key_identifier",
         @"nonce" : @"4a11a9cc-3bc3-11ec-8d3d-0242ac130003",
-        @"signature" : @"test_signature",
+        @"signature" : @YES,
         @"timestamp" : @(1635847102),
       }
     };
 
-    FIAPaymentQueueHandler *mockHandler = OCMClassMock(FIAPaymentQueueHandler.class);
-    id translator = OCMClassMock(FIAObjectTranslator.class);
+    TestPaymentQueueHandler *testHandler = [[TestPaymentQueueHandler alloc] init];
+
+    __block NSInteger addPaymentCount = 0;
+    testHandler.addPaymentStub = ^BOOL(SKPayment *_Nonnull payment) {
+      addPaymentCount++;
```

Comment:
Let's keep the naming convention consistent (`addPaymentCount` here vs `addPaymentTimes` previously)

Reply:
ill just change everything to be addFooCount